src/codegen/generators/expression_generator.rs

```diff
@@ -855,17 +899,74 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
                 self.generate_call_struct_argument_assignment(&CallParameterAssignment {
                     assignment_statement: exp,
                     function_name,
-                    parameter_type: None,
                     index: index as u32,
                     parameter_struct,
                 })?;
             if let Some(parameter) = parameter {
                 result.push(parameter.into());
             };
         }
+
         Ok(result)
     }
 
+    fn get_parameter_type(&self, parameter: &VariableIndexEntry) -> String {
+        if let Some(DataTypeInformation::Pointer {
+            inner_type_name,
+            auto_deref: true,
+            ..
+        }) = self
+            .index
+            .find_effective_type_info(parameter.get_type_name())
+        {
+            inner_type_name.into()
+        } else {
+            parameter.get_type_name().into()
+        }
+    }
+
+    fn generate_empty_expression(
+        &self,
+        parameter: &VariableIndexEntry,
+    ) -> Result<BasicValueEnum<'ink>, Diagnostic> {
+        let parameter_type_name = self.get_parameter_type(parameter);
+        let parameter_type = self.llvm_index.get_associated_type(&parameter_type_name)?;
+        match parameter.get_declaration_type() {
+            ArgumentType::ByVal(..) => {
+                if let Some(initial_value) =
+                    self.get_initial_value(&parameter.initial_value, &parameter_type_name)
+                {
+                    self.generate_expression(initial_value)
+                } else {
+                    let ptr_value = self.llvm.builder.build_alloca(parameter_type, "");
+                    Ok(self.llvm.load_pointer(&ptr_value, ""))
+                }
+            }
+            _ => {
+                let ptr_value = self.llvm.builder.build_alloca(parameter_type, "");
+
+                if let Some(initial_value) =
+                    self.get_initial_value(&parameter.initial_value, &parameter_type_name)
+                {
+                    let value = self.generate_expression(initial_value)?;
```

Comment:
Can you add an explanation of why we did this, like it's because we need the variable to be initialized even if it's a pointer..